junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/extension/DefaultLocaleExtension.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2015-2025 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v2.0 which
+ * accompanies this distribution and is available at
+ *
+ * https://www.eclipse.org/legal/epl-v20.html
+ */
+
+package org.junit.jupiter.engine.extension;
+
+import java.lang.reflect.AnnotatedElement;
+import java.util.Locale;
+
+import org.junit.jupiter.api.extension.AfterEachCallback;
+import org.junit.jupiter.api.extension.BeforeEachCallback;
+import org.junit.jupiter.api.extension.ExtensionConfigurationException;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.api.extension.ExtensionContext.Namespace;
+import org.junit.jupiter.api.util.DefaultLocale;
+import org.junit.jupiter.api.util.LocaleProvider;
+import org.junit.jupiter.api.util.LocaleProvider.NullLocaleProvider;
+import org.junit.jupiter.engine.support.JupiterLocaleUtils;
+import org.junit.platform.commons.support.AnnotationSupport;
+import org.junit.platform.commons.support.ReflectionSupport;
+
+class DefaultLocaleExtension implements BeforeEachCallback, AfterEachCallback {
+
+	private static final Namespace NAMESPACE = Namespace.create(DefaultLocaleExtension.class);
+
+	private static final String KEY = "DefaultLocale";
+
+	@Override
+	public void beforeEach(ExtensionContext context) {
+		AnnotatedElement element = context.getElement().orElse(null);
+		AnnotationSupport.findAnnotation(element, DefaultLocale.class).ifPresent(
```

Comment:
This won't find annotations on top-level and nested classes (which is why the test fail). You can either search the hierarchy upward (like the Pioneer implementation does) or implement `BeforeAllCallback` and store the value in the class-level store so it can be used as a fallback here.

Reply:
As discussed via email, I've pushed a commit that uses `BeforeAllCallback` to achieve this. The second commit moves the extension implementation to junit-jupiter-api to avoid its overhead since it should only be active when the `@DefaultLocale` annotation is used.

@Bukama Could you please do the same for `@DefaultTimeZone`?